src/sonic-config-engine/minigraph.py

```diff
@@ -403,6 +403,15 @@ def parse_deviceinfo(meta, hwsku):
                 if desc != None:
                     port_descriptions[port_alias_map.get(alias, alias)] = desc.text
                 port_speeds[port_alias_map.get(alias, alias)] = speed
+            interfaces = device_info.find(str(QName(ns, "ManagementInterfaces")))
+            if interfaces:
+                for interface in interfaces.findall(str(QName(ns1, "ManagementInterface"))):
+                    alias = interface.find(str(QName(ns, "InterfaceName"))).text
+                    speed = interface.find(str(QName(ns, "Speed"))).text
+                    desc  = interface.find(str(QName(ns, "Description")))
+                    if desc != None:
+                        port_descriptions[port_alias_map.get(alias, alias)] = desc.text
+                    port_speeds[port_alias_map.get(alias, alias)] = speed
```

Comment:
You may combine 2 loops into one.

Reply:
Which is the other loop you are referring to?